File: src/FactoryMethod/src/FilmeFactory.java
import java.util.ArrayList;

public class FilmeFactory implements MidiaFactory {
    @Override
    public Midia criarMidia(String titulo, String descricao, int anoLancamento, int duracao, String genero, String classificacao, ArrayList<String> elenco) {
        return new Filme(titulo, descricao, anoLancamento, duracao, genero, classificacao, elenco);
    }
}

File: src/FactoryMethod/src/factories/filme-factory.ts
import { MidiaFactory } from './midia-factory';
import { Filme } from '../models/filme';
import { Midia } from '../models/midia';

export class FilmeFactory implements MidiaFactory {
    public criarMidia(
        titulo: string,
        descricao: string,
        anoLancamento: number,
        duracao: number,
        genero: string,
        classificacao: string,
        elenco: string[],
        diretor: string,
        roteirista: string
    ): Midia {
        return new Filme(titulo, descricao, anoLancamento, duracao, genero, classificacao, elenco, diretor, roteirista);
    }
}
